### PR TITLE
Add package descriptions for transport, netty and okhttp

### LINKED ISSUE #254: Document transport layers (netty vs okhttp)
Hi, am I right that there are currently two implementations of the transport layer? Netty and okhttp? I was curious if there are any situations in which one might be preferable to the other as I'm not quite sure why there are two implementations.

### PR DESCRIPTION
**Pull request by zhangkun83**
Resolves #254 #426

### CODE REVIEW

**Comment by ejona86** [comment]
@zhangkun83 LGTM

**Comment by ejona86** [comment]
I don't think this alone solves #254; I think that would need an update to README.md, or some other more-visible documentation.

**Comment by zhangkun83** [comment]
@ejona86 I have updated the README. PTAL.

File: README.md [hunk 1]
```diff
@@ -265,7 +273,24 @@ The 'channel' layer is an abstraction over transport handling that is suitable f
...
+The Transport layer does the heavy lifting of putting & taking bytes off the
+wire. The interfaces to it are abstract just enough to allow plugging in of
+different implementations. Transports are modeled as `Stream` factories. The
+variation in interface between a server Stream and a client Stream exists to
+codify their differing semantics for cancellation and error reporting.
+
+Note the transport layer API is considered internal to gRPC and has weaker API
+guarantees than the core API under package `io.grpc`.
+
+gRPC comes with two Transport implementations:
```
Comment (ejona86) [inline README.md]: Three? Mention InProcess for when a server may be local, and for testing? I guess not necessary now.

**Comment by ejona86** [comment]
@zhangkun83 LGTM

File: README.md (hunk 1 above)
Comment (zhangkun83) [inline README.md]: I added it anyway. This is a feature that some people will look for and it's nicer to expose it here.

**Pull request merged**